deferral/ah-django
# #164859749 User reset password via email

## What does this PR do?
- Creates user reset password endpoint

## Description of Task to be completed?
- [x] Create user reset password endpoint

## How should this be manually tested?
- Clone the repo to your local machine from here

- cd into the ah-django directory and checkout to the branch `ft-reset-password-viaemail-164859749`

- `cd ah-django`

- git checkout `ft-reset-password-viaemail-164859749`

- Create a virtual environment and activate it:

- `python3 -m virtualenv venv && source venv/bin/activate`

- If you need to install virtualenv first:

- `pip install virtualenv`

- Install the project dependencies

- `pip install -r requirements.txt`

- Make database migrations and start the django server

- `python3 manage.py makemigrations`

- `python3 manage.py migrate`

- `python3 manage.py runserver`

- signup as a new user using a valid email

- try changing passing using the signed up email as the input

- check your inbox to find the send link and click on it to reset the password

- for `receiving a reset link` provide the data to postman in this format:
  ```  
     {
	 "email": "test.user@gmail.com"
     }
  ```
- for `resetting the password` provide the data to postman in this format:
  ```
     {
       "password": "mYp@55ord"
       "password": "mYp@55ord"
     }
  ```
- request type `POST`
## Any background context you want to provide?
- Initially, the application had no reset password endpoint

## What are the relevant pivotal tracker stories?
#164859749

## Screenshots:
![Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_161](https://user-images.githubusercontent.com/42172425/55416740-0c299900-5578-11e9-9ac6-93587f22487d.png)
![Selection_160](https://user-images.githubusercontent.com/42172425/55416770-18adf180-5578-11e9-811b-06180b764235.png)
![Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_166](https://user-images.githubusercontent.com/42172425/55468495-2b720600-560c-11e9-9cf6-22032540fc84.png)
![Selection_164](https://user-images.githubusercontent.com/42172425/55468513-388ef500-560c-11e9-89e6-23ded3804275.png)
